File: CarePackage/Software/MozillaFirefox.cs
namespace CarePackage.Software;

public class MozillaFirefox : BaseSoftware
{
    public override string           Key            { get; set; } = "mozilla-firefox";
    public override string           Name           { get; set; } = "Mozilla Firefox";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Browser;
    public override string           DownloadName   { get; set; } = "Firefox Installer.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.mozilla_firefox;
    public override string           Homepage       { get; set; } = "https://mozilla.org/firefox";
    
    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://download.mozilla.org/?product=firefox-latest-ssl&os=win64&lang=en-US");
}

File: packages/app/src/services/software/definitions/mozillaFirefox.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'mozilla-firefox';
	public name = 'Mozilla Firefox';
	public category = [SoftwareCategory.Browser];
	public downloadName = 'Firefox Installer.exe';
	public isArchive = false;
	public shouldCacheUrl = false;
	public requiresAdmin = false;
	public deprecated = false;
	public icon = 'mozilla-firefox.png';
	public homepage = 'https://mozilla.org/firefox';

	public async resolveDownloadUrl() {
		return ok('https://download.mozilla.org/?product=firefox-latest-ssl&os=win64&lang=en-US');
	}
}
